x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/es_index.tsx

```diff
@@ -38,33 +70,165 @@ export function getActionType(): ActionTypeModel {
 
 const IndexActionConnectorFields: React.FunctionComponent<ActionConnectorFieldsProps<
   EsIndexActionConnector
->> = ({ action, editActionConfig }) => {
-  const { index } = action.config;
+>> = ({ action, editActionConfig, errors, http }) => {
+  const { index, refresh, executionTimeField } = action.config;
+  const [hasTimeFieldCheckbox, setTimeFieldCheckboxState] = useState<boolean>(
+    executionTimeField !== undefined
+  );
+
+  const [indexPatterns, setIndexPatterns] = useState([]);
+  const [indexOptions, setIndexOptions] = useState<EuiComboBoxOptionOption[]>([]);
+  const [timeFieldOptions, setTimeFieldOptions] = useState([firstFieldOption]);
+  const [isIndiciesLoading, setIsIndiciesLoading] = useState<boolean>(false);
+
+  useEffect(() => {
+    const indexPatternsFunction = async () => {
+      setIndexPatterns(await getIndexPatterns());
+      if (index) {
+        const currentEsFields = await getFields(http!, [index]);
+        const timeFields = getTimeFieldOptions(currentEsFields as any);
+        setTimeFieldOptions([firstFieldOption, ...timeFields]);
+      }
+    };
+    indexPatternsFunction();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
   return (
-    <EuiFormRow
-      fullWidth
-      label={i18n.translate(
-        'xpack.triggersActionsUI.components.builtinActionTypes.indexAction.indexTextFieldLabel',
-        {
-          defaultMessage: 'Index (optional)',
-        }
-      )}
-    >
-      <EuiFieldText
+    <>
+      <EuiHorizontalRule />
```

Comment:
Is this horizontal rule intentional? It seemed strange in the UI (above "Write to index").

<img width="1247" alt="Screen Shot 2020-03-17 at 9 18 38 AM" src="https://user-images.githubusercontent.com/3694571/76859770-51e67c80-6830-11ea-8433-204e8c679ab0.png">

Reply:
did it like the mockups has. But agree with you - better to remove it to stay consistent with other connectors UI
![Img](https://i.imgur.com/28GK1Jx.png)